Editor/ProjectAuditorSettingsProvider.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Unity.ProjectAuditor.Editor
+{
+    /// <summary>
+    /// A built-in provider to manage ProjectAuditorSettings default and user-generated assets for customization.
+    /// </summary>
+    public class ProjectAuditorSettingsProvider : IProjectAuditorSettingsProvider
```

Comment:
> Do we need to worry about domain-reload? Does everything persist?

Oh, right, since the params again also include the platform. I'll do that!

Reply:
> Since some analyzers require platform as well, I would as further as passing the ProjectAuditorParams to 

Right, makes sense with params already including the platform. I'll do that!

> Do we need to worry about domain-reload? Does everything persist?

Let me do one more pass to test this and think about where this could have undesired results.

The most suspicious part of the code that may have a flaw is the PA window, if that window was open during reload (since it keeps an instance of the provider).